backend/metadata_writer/src/metadata_writer.py

```diff
@@ -164,8 +169,17 @@ def is_kfp_v2_pod(pod) -> bool:
             pod_name = obj.metadata.name
 
             # Logging pod changes for debugging
-            with open('/tmp/pod_' + obj.metadata.name + '_' + obj.metadata.resource_version, 'w') as f:
+            debug_path = '/tmp/pod_' + obj.metadata.name + '_' + obj.metadata.resource_version
+            with open(debug_path, 'w') as f:
                 f.write(yaml.dump(obj.to_dict()))
+            debug_paths.append(debug_path)
+
+            # Do some housekeeping, ensure we only keep a fixed size buffer of debug files so we don't
+            # grow the disk size indefinitely for long running pods.
+            debug_paths_to_remove = debug_paths[:-debug_files_size]
+            debug_paths = debug_paths[-debug_files_size:]
+            for debug_path in debug_paths_to_remove:
+                os.remove(debug_path)
```

Comment:
If I'm not mistaken, this chunk of code could be replaced with the following that does the same but much more readable:
```python
if len(debug_paths) > debug_file_size:
    os.remove(debug_paths.pop(0))
```

I'd suggest future improvement using `collections.deque` for constant time on pop operation.
https://wiki.python.org/moin/TimeComplexity

WDYT?

Reply:
Perfect, looks like the syntax is `.popleft()` to pop from the front of the queue but you're correct, made the amendments. Thanks for all the help making this more efficient, got me wondering how inefficient my code is normally now 😅